src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -21,27 +21,40 @@ namespace Voron.Impl
     public unsafe class LowLevelTransaction : IDisposable
     {
         private const int PagesTakenByHeader = 1;
+
         public readonly AbstractPager DataPager;
         private readonly StorageEnvironment _env;
         private readonly long _id;
         private readonly ByteStringContext _allocator;
         private readonly bool _disposeAllocator;
-        private Tree _root;
 
-        internal FixedSizeTree _freeSpaceTree;
+        private Tree _root;
+        public Tree RootObjects => _root;        
 
-        public bool FlushedToJournal;
-        public Tree RootObjects => _root;
+        public bool FlushedToJournal;        
 
         private readonly WriteAheadJournal _journal;
+        internal readonly List<JournalSnapshot> JournalSnapshots = new List<JournalSnapshot>();
 
+        private static ObjectPool<Dictionary<long, PageFromScratchBuffer>, DictionaryResetBehavior<long, PageFromScratchBuffer>> ScratchPagesTablePool = new ObjectPool<Dictionary<long, PageFromScratchBuffer>, DictionaryResetBehavior<long, PageFromScratchBuffer>>(() => new Dictionary<long, PageFromScratchBuffer>(NumericEqualityComparer.Instance), 30);
```

Comment:
Why is this static? Isn't it better to do it at the level of the Environment?

My concern is that it is quite easy to have more than 30 storage environments, and then we'll have a lot of churn.